tests/test_walker.py

```diff
@@ -38,8 +38,24 @@ def test_autoreducing():
     filelist = [frame.filepath for frame in l1_dataset]
 
 
+    ###########################################
+    ### Create a dummy non-linearity file ####
+    #Create a mock dataset because it is a required input when creating a NonLinearityCalibration
+    dummy_dataset = mocks.create_prescan_files()
+
+    # Make a non-linearity correction calibration file
+    input_non_linearity_filename = "nonlin_table_TVAC.txt"
+    input_non_linearity_path = os.path.join(os.path.dirname(__file__), "test_data", input_non_linearity_filename)
+    test_non_linearity_filename = input_non_linearity_filename.split(".")[0] + ".fits"
+    nonlin_fits_filepath = os.path.join(os.path.dirname(__file__), "test_data", test_non_linearity_filename)
+    tvac_nonlin_data = np.genfromtxt(input_non_linearity_path, delimiter=",")
+
+    pri_hdr, ext_hdr = mocks.create_default_headers()
+    new_nonlinearity = data.NonLinearityCalibration(tvac_nonlin_data,pri_hdr=pri_hdr,ext_hdr=ext_hdr,input_dataset = dummy_dataset)
+    new_nonlinearity.filename = nonlin_fits_filepath
+    new_nonlinearity.save()
     # index the sample nonlinearity correction that we need for processing
-    new_nonlinearity = data.NonLinearityCalibration(os.path.join(os.path.dirname(__file__),"test_data",'nonlin_sample.fits'))
+    # new_nonlinearity = data.NonLinearityCalibration(os.path.join(os.path.dirname(__file__),"test_data",'nonlin_sample.fits'))
```

Comment:
did you mean to remove this line?

Reply:
Yes. This is because we've already created a new_nonlinearity cal a few lines before this. We don't need to read it in again, right?

Reply:
Yeah sorry, I meant did you mean to delete it, instead of commenting it out (since you deleted the rest of them in this file).

Reply:
Ok deleted now.